public/assets/js/components/Life.js

```diff
@@ -10,6 +10,8 @@ class Life {
 
   constructor(lifeNodeList) {
     this.lifeNodeList = lifeNodeList;
+    // thought we should initialize the counter too
+    this.counter = this.counterRange[1];
```

Comment:
This is not wrong. But just so to be aware of it, you could also initialize the counter outside of constructor. 
It'd look similar to something like:
```javascript
class Life {
  counterRange = [0, 3];

  #counter = this.counterRange[1]; // I used the previous property to give this one a default value.

  ... (other stuff)
```

Again, I think it's totally a matter of personal preference. 👍